File: backend/app/schemas/notice.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict, field_validator


class NoticeCreate(BaseModel):
    pool_id: Optional[int] = None
    category: str
    title: str
    content: str


class NoticeUpdate(BaseModel):
    reason: str

    @field_validator("reason")
    @classmethod
    def reason_not_empty(cls, v: str) -> str:
        if not v.strip():
            raise ValueError("수정 사유를 입력해주세요.")
        return v.strip()

    pool_id: Optional[int] = None
    category: Optional[str] = None
    title: Optional[str] = None
    content: Optional[str] = None


class NoticeResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    pool_id: Optional[int] = None
    category: Optional[str] = None
    title: str
    has_attachment: bool = False
    created_by_name: Optional[str] = None
    created_at: Optional[datetime] = None


class NoticeDetail(NoticeResponse):
    content: Optional[str] = None
    attachment_doc_id: Optional[int] = None
    attachment_name: Optional[str] = None


class NoticeListResponse(BaseModel):
    items: list[NoticeResponse]
    total: int
    page: int
    size: int

File: frontend/src/types/notice.ts
export interface Notice {
  id: number;
  category: string;
  title: string;
  has_attachment: boolean;
  created_by_name: string;
  created_at: string;
}

export interface NoticeFileItem {
  id: number;
  file_name: string;
  file_size: number | null;
}

export interface NoticeDetail extends Notice {
  content: string;
  attachment_doc_id: number | null;
  attachment_name: string | null;
  files: NoticeFileItem[];
}

export interface NoticeCreateInput {
  pool_id?: number;
  category: string;
  title: string;
  content: string;
  file?: File;
}

export interface NoticeListResponse {
  items: Notice[];
  total: number;
}
